### DISCUSSION TITLE
zh/guide/advanced/sso

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#173] [Announcements]
# zh/guide/advanced/sso

Single sign-on client id 客户端ID Single sign-on client secrets 秘钥ID 注意事项 1. AList 后台参数也记得写好保存，写好保存后也要回到个人资料下方会有一个需要绑定的按钮进行绑定，否则无法使用 2. 四种单点登录暂时只能启用一种，如果你之前用的是别的单点登录方式然后换了新的单点登录方式 ...

https://alist.nn.ci/zh/guide/advanced/sso.html

**Comment by chuyiyue** [comment]
麻烦问一下比如说一个账号绑定钉钉之后怎么解绑

**Reply by anwen-anyi** [reply]
AList单点登录配置好了，然后后台个人资料哪里绑定后不是可以解绑吗？
如果还觉得不放心 可以去钉钉后台删除应用或者把回调参数删了呗~甚至直接将整个应用删除
以上的回答希望对你有帮助

**Reply by chuyiyue** [reply]
绑定后没有解绑按钮![image.png](https://s2.loli.net/2023/03/07/k9bjHamZOdyS4Gr.png)

**Reply by anwen-anyi** [reply]
你只是在单点登录哪里配置好了吧？ 你还没绑定这个单点登录平台吖？

**Reply by chuyiyue** [reply]
扫码绑定了的，可以正常扫码登录，但是不管怎么扫码绑定，都会绑定到管理员admin账户，钉钉扫码之后也是登录管理员admin，所以我想解绑之后重新绑定一下

**Reply by anwen-anyi** [reply]
好像就是只能绑定admin吧？你指的想绑定 非admin账号密码

**Reply by chuyiyue** [reply]
是的，我以为非admin用户也是可以绑定的。当然，如果只能绑定admin账户也是可以的，我还以为是我操作的问题呢

**Reply by chuyiyue** [reply]
谢谢你的解答

**Reply by anwen-anyi** [reply]
现在应该只能绑定admin账号应该，大概似乎也许没考虑非admin账号以外的

**Reply by anwen-anyi** [reply]
你试试用admin吧 测试的时候 admin账户是没问题的~

**Reply by Vito-huanghk** [reply]
麻烦问下如何限制别的其他公司的钉钉登录呢？ 是不是只要是钉钉就能扫码登录？

**Comment by Connor3773** [comment]
按照文档配置好之后点击绑定就会有这个错误redirect_uri_mismatch。不知道是什么问题，平台用的GitHub

**Reply by anwen-anyi** [reply]
你在github的回调参数写的啥

**Reply by Connor3773** [reply]
https://你的域名/api/auth/sso_callback
按教程里的，写的这个

**Reply by Connor3773** [reply]
GitHub点完授权之后小窗白屏，然后页面url变成这个
```
https://xxx.com/api/auth/sso_callback?error=redirect_uri_mismatch&error_description=The+redirect_uri+MUST+match+the+registered+callback+URL+for+this+application.&error_uri=https%3A%2F%2Fdocs.github.com%2Fapps%2Fmanaging-oauth-apps%2Ftroubleshooting-authorization-request-errors%2F%23redirect-uri-mismatch
```

**Reply by anwen-anyi** [reply]
> https://你的域名/api/auth/sso_callback
> 按教程里的，写的这个

你的域名 你写的你自己的域名还是写的这个中文😂

截图看看你单点登录哪里的配置  
还有就是你在个人中心哪里绑定了吗?
你查查看呢写的没问题 教程发的也没问题😂

**Reply by Connor3773** [reply]
是这样配置的
![](https://img.bitpd.com/i/2023/03/11/640ca576e1d3f.webp)
![](https://img.bitpd.com/i/2023/03/12/640ca5b590fcd.webp)
![](https://img.bitpd.com/i/2023/03/12/640ca6061a7ff.webp)

**Reply by Connor3773** [reply]
解决了。。。我把回调url的https改成了http就好了，谢谢回复！

**Reply by anwen-anyi** [reply]
> 解决了。。。我把回调url的https改成了http就好了，谢谢回复！

呃  你的站点没有HTTPS么QAQ

**Reply by Connor3773** [reply]
我的站点是强制HTTPS的，出现这问题很奇怪。
之前最初版单点登录GitHub回调填HTTPS都是正常的（就是回调链接结尾还是github_callback的那时候），升级后不行了。

**Reply by anwen-anyi** [reply]
好吧…不过现在解决了就行😂
后续也加一个说明补充一下

**Reply by Connor3773** [reply]
嗯嗯好的，谢谢你:)

**Comment by InoryS** [comment]
希望可以支持 keycloak。
Keycloak是一个开源软件产品，旨在为现代的应用程序和服务，提供包含身份管理和访问管理功能的单点登录工具。
支持 OIDC 和 SAML 协议。

也就是支持自定义填写 Authorize URL、Token URL、Userinfo URL、Logout URL、Userinfo ID 即可。
同时相当于支持其他 OIDC 身份提供者。

**Comment by azathoth3** [comment]
输入重定向URL的时候Azure提示：
"无法更新应用程序。错误详细信息: URL 不得包含查询字符串。 [9X4jfFyWKwulrng5smMIpE]"

**Reply by yws112358** [reply]
这个查了微软给的资料，你肯定开启了微软个人账户的，选择了这个链接不能包含？这种字符，所以有些用不上了

**Comment by alanhe421** [comment]
单点登录页面改版了，所以![https://i.imgur.com/DxRxaK6.jpg](https://i.imgur.com/DxRxaK6.jpg)

现在是只需要配置上面几项是吗？下面的也得配置了吧

**Reply by alanhe421** [reply]
解决了，红色部分对应把ID，Secret也填写下就行了。这里重复了。

**Reply by anwen-anyi** [reply]
这些红色的是给 `Casdoor` 填写的，不过上面的也是需要填写，如果你不需要 `Casdoor `只把上面两个填写一下就行其它的无视即可

**Reply by alanhe421** [reply]
实际测试，只写下面的即`Casdoor `的可以，只写上面的如下图我没成功。

![image](https://github.com/alist-org/docs/assets/9245110/14a51858-51fb-4128-a16a-7f52b5fcf4c0)


## version
v3.18.0

**Comment by wintbiit** [comment]
可以实现单点登录自动注册吗

**Reply by anwen-anyi** [reply]
单点登录可以注册，但是不能自动注册AList用户

**Comment by LegendAdmin** [comment]
我看了视频教程之后才知道这个“单点登录”原来就是指“一键登录”啊😂我还一直很费解

**Comment by bbxblue** [comment]
使用微软的e5账号配置后，貌似整个组织的用户都可以访问绑定单点登录的管理员账号，是我的步骤不对还是别的问题？我该怎么只允许指定的一个微软账号登录？

**Reply by bbxblue** [reply]
已解决

**Comment by zyyfriends** [comment]
请问一下，使用群晖SSO服务器，选择OIDC协议。配置完后，绑定单点登录平台，提示{"code":400,"message":"no method provided","data":null}
看服务器日志，实际已经返回了登录成功
如图https://pic.zhouyuyang.com/alistError.png
![](https://pic.zhouyuyang.com/alistError.png)

**Reply by assassin1314** [reply]
兄弟，我搞定了，可以交流一下，配置有点复杂，我还是在你这个截图得到的启发

**Reply by zyyfriends** [reply]
麻烦看一下你的配置呢

**Reply by chenjuefei** [reply]
https://github.com/alist-org/alist/discussions/4870

**Comment by Vito-huanghk** [comment]
麻烦问下如何限制别的其他公司的钉钉登录呢？ 是不是只要是钉钉就能扫码登录？

**Comment by 66508393** [comment]
可以增加 企业微信 的单点登陆吗？

**Reply by anwen-anyi** [reply]
> 可以增加 企业微信 的单点登陆吗？
> 

可以使用Casdoor来进行配置

**Reply by 66508393** [reply]
好的 谢谢
